external-crates/move/crates/move-analyzer/trace-adapter/src/runtime.ts

```diff
@@ -197,14 +199,101 @@ interface IRuntimeStackFrame {
 }
 
 /**
- * Describes the runtime stack during trace viewing session
+ * Describes the Move call stack during trace viewing session
  * (oldest frame is at the bottom of the stack at index 0).
  */
-export interface IRuntimeStack {
-    frames: IRuntimeStackFrame[];
+export interface IMoveCallStack {
+    frames: IMoveCallStackFrame[];
     globals: Map<number, RuntimeValueType>;
 }
 
+/**
+ * Describes information about a Move call.
+ */
+export interface IMoveCallInfo {
+    pkg: string;
+    module: string;
+    function: string;
+}
+
+/**
+ * Describes the type of PTB summary.
+ */
+export type PTBSummary = IMoveCallInfo | string;
+
+/**
+ * Describes PTB frame containing the PTB summary.
+ */
+interface IPTBSummaryFrame {
+    /**
+     * PTB summary frame ID.
+     */
+    id: number;
+    /**
+     * PTB summary frame (virtual) line.
+     */
+    line: number;
+    /**
+     * PTB summary.
+     */
+    summary: PTBSummary[];
+}
+
+/**
+ * Kind of native PTB command (other tham Move call).
+ * Strings reflect names of the commands in the PTB.
+ */
+export enum PTBCommandKind {
+    MoveCallStart = 'MoveCallStart', // used in trace to identify Move call start
+    MoveCallEnd = 'MoveCallEnd', // used in trace to identify Move call end
+    SplitCoinsStart = 'SplitCoins', // used as commad name in UI
+    SplitCoinsEnd = 'SplitCoinsEnd',
+    TransferStart = 'Transfer', // used as command name in UI
+    TransferEnd = 'TransferEnd',
+}
+
+/**
+ * Contains information about a PTB command stack frame.
+ */
+export type PTBNativeCommandStackFrameInfo =
+    | {
+        kind: PTBCommandKind.TransferStart
+        description: string
+        locals: IRuntimeVariable[]
+    } | {
+        kind: PTBCommandKind.SplitCoinsStart
+        description: string
+        locals: IRuntimeVariable[]
+    };
+
+/**
+ * Describes PTB command stack frame.
+ */
+export interface IPTBCommandStackFrame {
+    id: number;
+    line: number;
+    command: PTBNativeCommandStackFrameInfo;
+}
+
+/**
+ * Describes PTB call staack during trace viewing session.
+ */
+export interface IPTBStack {
```

Comment:
The reason both fields are optional is to maintain compatibility with unit test tracing where we will only have `commandFrame` holding Move execution stack. On the other hand, we may also have just `summaryFrame` present at the beginning of a PTB debugging session